src/cic/model.clj

```diff
@@ -213,6 +205,56 @@
                 (let [next-placement (phase-transition (zero? offset) age placement)]
                   (recur next-offset next-placement (conj placements {:offset next-offset :placement next-placement})))))))))))
 
+(defn phase-durations
+  "Calculate the phase durations for all closed periods"
+  [periods]
+  (let [durations (mapcat (fn [{:keys [birthday beginning end episodes]}]
```

Comment:
pull out the fn into something separate, and I *think* this can be rewritten as

```
(into []
  (comp
    (remove :open?)
    (mapcat pulled-out-fn))
  periods)
```